ouroboros-consensus/src/Ouroboros/Storage/ChainDB/API.hs

```diff
@@ -424,7 +423,6 @@ data StreamTo blk =
 data Iterator m blk b = Iterator {
       iteratorNext  :: m (IteratorResult blk b)
     , iteratorClose :: m ()
```

Comment:
Add a comment saying that an `fmap`-ed or traversed iterator still refers to the original one, so closing either will close both.